app/api/dao/user.py

```diff
@@ -148,11 +148,8 @@ def list_users(user_id, is_verified=None):
         users_list = UserModel.query.filter(UserModel.id != user_id).all()
         list_of_users = []
         if is_verified:
-            for user in users_list:
-                if user.is_email_verified:
-                    list_of_users += [user.json()]
-        else:
-            list_of_users = [user.json() for user in users_list]
+            users_list = filter(lambda user: user.is_email_verified, users_list)
+        list_of_users = [user.json() for user in users_list]
```

Comment:
Didn't use map() and lambda for this statement because it maps blank fields as well. This causes tests to fail. Also I feel that existing ```list_of_users = [user.json() for user in users_list]``` is neat enough.

```py
list_of_users = list(map(lambda user: user.json, users_list))
```

Output
```sh
(venv) home@HP:~/Desktop/Dak/mentorship-backend$ python -m unittest discover tests
/home/shardul/Desktop/Dak/mentorship-backend/venv/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py:374: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
  args = inspect.formatargspec(*spec)
/home/shardul/Desktop/Dak/mentorship-backend/venv/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py:384: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
  spec[2], None, spec[4])
/home/shardul/Desktop/Dak/mentorship-backend/venv/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py:405: DeprecationWarning: `formatargspec` is deprecated since Python 3.5. Use `signature` and the `Signature` object directly
  formatvalue=lambda x: '=' + x)
................................................................................................................F.F........................................................................
======================================================================
FAIL: test_list_users_api_resource_auth (users.test_api_list_users.TestListUsersApi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/shardul/Desktop/Dak/mentorship-backend/tests/users/test_api_list_users.py", line 51, in test_list_users_api_resource_auth
    self.assertEqual(expected_response, json.loads(actual_response.data))
AssertionError: Lists differ: [{'id': 2, 'username': 'user1', 'name': 'User', '[417 chars]lse}] != [{'id': None, 'username': None, 'name': None, 'sl[408 chars]one}]

First differing element 0:
{'id': 2, 'username': 'user1', 'name': 'User', '[180 chars]alse}
{'id': None, 'username': None, 'name': None, 'sl[176 chars]None}

Diff is 1228 characters long. Set self.maxDiff to None to see it.

======================================================================
FAIL: test_list_users_api_resource_verified_users (users.test_api_list_users.TestListUsersApi)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/shardul/Desktop/Dak/mentorship-backend/tests/users/test_api_list_users.py", line 59, in test_list_users_api_resource_verified_users
    self.assertEqual(expected_response, json.loads(actual_response.data))
AssertionError: Lists differ: [{'id': 2, 'username': 'user1', 'name': 'User', '[181 chars]lse}] != [{'id': None, 'username': None, 'name': None, 'sl[177 chars]one}]

First differing element 0:
{'id': 2, 'username': 'user1', 'name': 'User', '[180 chars]alse}
{'id': None, 'username': None, 'name': None, 'sl[176 chars]None}

- [{'available_to_mentor': False,
?                          ^^^^

+ [{'available_to_mentor': None,
?                          ^^^

    'bio': None,
-   'id': 2,
?         ^

+   'id': None,
?         ^^^^

    'interests': None,
    'location': None,
-   'name': 'User',
?           ^^^ --

+   'name': None,
?           ^^^

-   'need_mentoring': False,
?                     ^^^^

+   'need_mentoring': None,
?                     ^^^

    'occupation': None,
    'organization': None,
    'skills': None,
    'slack_username': None,
-   'username': 'user1'}]
?               ^^^ ---

+   'username': None}]
?               ^^^


----------------------------------------------------------------------
Ran 187 tests in 15.825s

FAILED (failures=2)
```